### ISSUE TITLE
[HOLD for payment 2023-05-03] [$1000] Chat - "You appear to be offline" is misaligned

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#16838]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
or open the App
2. Login with any account
3. Go to any chat
4. Disable the internet connection

## Expected Result:
Message "You appear to be offline" is equally centered under compose box in all environments.

## Actual Result:
Message "You appear to be offline" is misaligned in mWeb and App.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.94.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![Screenshot 2023-04-01 030015-1](https://user-images.githubusercontent.com/93399543/229295986-10479f36-70e8-44ce-973a-07aabc1a947c.png)

![Screenshot 2023-04-01 030108-2](https://user-images.githubusercontent.com/93399543/229295987-9630a959-209a-4ea0-a97b-63c358d71e03.png)

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012997ad359b23a2f5</li>
        <li>Upwork Job ID: 1645861791230992384</li>
        <li>Last Price Increase: 2023-04-18</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by alexpensify** [comment]
I was assigned while I was OOO, I'll try to test tomorrow.

**Comment by alexpensify** [comment]
I'm able to replicate: 
![image](https://user-images.githubusercontent.com/8107070/230214461-e55ce239-ccb1-4124-9f03-db331fb9bbe5.png)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @chiragsalian (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by alexpensify** [comment]
@chiragsalian - I've assigned to Engineering for extra eyes before assigning the External label. Thanks for the review.

**Comment by chiragsalian** [comment]
I don't mind changing this but i think @shawnborton preferred it left align wrt the compose box. Shawn can you verify and suggest changes if needed. Once we lock on a design we can add external label.

**Comment by shawnborton** [comment]
I agree, it would be nice to vertically center it within the given space that is available.

**Comment by alexpensify** [comment]
@shawnborton - to confirm, this needs to centered or move over to be under the + sign half circle? Thanks! 

![image](https://user-images.githubusercontent.com/8107070/230648962-4f96157f-53ea-4715-969d-2c46b1ef18cc.png)

**Comment by shawnborton** [comment]
I think it just need to be vertically centered within that space it occupies under the chat composer but above the home bar. So basically just keep the composer where it is, but move the message down just a tad.

**Comment by alexpensify** [comment]
@chiragsalian - I think we have the information we need here. Let me know if you agree and we can add the External label.

**Comment by chiragsalian** [comment]
oh vertically aligned. Cool cool.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~012997ad359b23a2f5

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - "You appear to be offline" is misaligned

### What is the root cause of that problem?
We are rendering the  `OfflineIndicator` Component independently between small screen (mWeb, native) and big screen (web, desktop) 

In Web and desktop, we apply this style for `OfflineIndicator` Component 
https://github.com/Expensify/App/blob/56c27de50093a85c1c25ad5ac359b043c67ab96b/src/styles/styles.js#L718-L722
In mWeb and native, we apply this style `OfflineIndicator` Component 
https://github.com/Expensify/App/blob/56c27de50093a85c1c25ad5ac359b043c67ab96b/src/styles/styles.js#L739-L743

Because we are setting padding-bottom: 9px and without paddingTop, It causes this bug
### What changes do you think we should make in order to solve the problem?
We should replace `padding-bottom: 9px` to `padding-top: 5px and padding-bottom: 5px OR margin-top: 5px and margin-bottom: 5px`


### What alternative solutions did you explore? (Optional)
NA

### Result
<img width="320" alt="Screen Shot 2023-04-05 at 17 33 28" src="https://user-images.githubusercontent.com/129500732/230056102-a45d3862-26ed-4ac4-8018-9b0f83d9dc6b.png">

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - "You appear to be offline" is misaligned in Android Native and Chrome, iOS Native and Safari

### What is the root cause of that problem?
For desktop we display `<OfflineIndicator>` via line 896 and passing `chatItemComposeSecondaryRow` style.
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/pages/home/report/ReportActionCompose.js#L896
This is `chatItemComposeSecondaryRow` style info.
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/styles/styles.js#L718-L722

So now `<OfflineIndicator>` component will render style via this
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/OfflineIndicator.js#L34-L39

So if `containerStyles` comes via props it will render it, so during desktop it will come, but for `isSmallScreenWidth ` like Android Native/Chrome and iOS Native/Safari there is no `containerStyles` array received, so it will render style via line 38 i.e. `styles.offlineIndicatorMobile`

Here is `offlineIndicatorMobile` style:
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/styles/styles.js#L739-L742

Here we can see this style not match with `chatItemComposeSecondaryRow`, So desktop and mobile device style not match. This is the root cause of the problem.

### What changes do you think we should make in order to solve the problem?
We have to adjust style within `offlineIndicatorMobile` such that it will match with `chatItemComposeSecondaryRow` style as shown below:
```javascript
offlineIndicatorMobile: { 
     paddingLeft: 20, 
     // paddingBottom: 9,    // *** Remove
     height: 15,            // ***  Add this
     marginBottom: 5,      // ***  Add this
     marginTop: 5,         // ***  Add this
 }, 
```
So this will solve the problem and "You appear to be offline" looks same everywhere as shown in results.

### What alternative solutions did you explore? (Optional)
None

<details>
<summary>Results</summary>

#### iOS Native
<img width="481" alt="iOS" src="https://user-images.githubusercontent.com/7823358/229365399-c5ebee2e-25b0-4f14-bc60-ab38da7d427f.png">

#### iOS Safari
<img width="461" alt="iOS-Safari" src="https://user-images.githubusercontent.com/7823358/229365413-cc52bf73-fa82-4d23-8bf8-14ea5801ecba.png">

#### Android Native
<img width="556" alt="Android" src="https://user-images.githubusercontent.com/7823358/229365480-c834076d-b236-4b17-9f7e-4ff8ff1520b3.png">

#### Android Chrome
<img width="556" alt="Android-Chrome" src="https://user-images.githubusercontent.com/7823358/229365504-e535552c-e52e-444e-8855-dac7fd9bc310.png">

#### MacOS Web
<img width="461" alt="MacOS-Web" src="https://user-images.githubusercontent.com/7823358/229365522-6d4ea103-6732-4718-8c1d-2892c8b51be0.png">

</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The offline indicator spacing is different between small screen and large screen.

### What is the root cause of that problem?
On small screen, the offline indicator comes from `ScreenWrapper` while on large screen it comes from `ReportActionCompose`.
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/ScreenWrapper/index.js#L123-L125
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/pages/home/report/ReportActionCompose.js#L890-L899

If you look at the code above, we have a container styles of `chatItemComposeSecondaryRow` which will be used as the `OfflineIndicator` container style.

https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/OfflineIndicator.js#L34-L39
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/styles/styles.js#L718-L722

But the style for small screen comes from 
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/styles/styles.js#L739-L742

This makes the style inconsistent.

### What changes do you think we should make in order to solve the problem?

I think we should apply the `chatItemComposeSecondaryRow` style as the default style for both small & large screen.
https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/OfflineIndicator.js#L34-L39

The returned style will be 
```
if (containerStyles.length) { 
    return containerStyles; 
} 
return [
    isSmallScreenWidth ? styles.pl5 : {},
    styles.chatItemComposeSecondaryRow,
];
```

We don't need `offlineIndicator` & `offlineIndicatorMobile` style anymore, but we still need the 20 padding left (pl5) for small screen.

Last, we should remove the `containerStyles` from the `OfflineIndicator`.

Note: this style change will affect all offline indicator on all pages

**Comment by alexpensify** [comment]
@mananjadhav and @chiragsalian - any thoughts on the proposals so far?

**Comment by mananjadhav** [comment]
I think all the proposals solve the problem, but I am going to use @dukenv0307's [proposal](https://github.com/Expensify/App/issues/16838#issuecomment-1503920274) for fixing the styles. It is simple and doesn't need a lot of refactor. 

That being said I would appreciate if @chiragsalian can also review other proposals. These proposals [here](https://github.com/Expensify/App/issues/16838#issuecomment-1503920275) and [here](https://github.com/Expensify/App/issues/16838#issuecomment-1504491031) make a case for refactor, but I am not sure we're at a point where we would want to add props like `isOfflineIndicatorRequired` or  remove the `containerStyles` props.

**Comment by alexpensify** [comment]
@chiragsalian - can you please review @mananjadhav feedback? Thanks!

**Comment by chiragsalian** [comment]
I like @dukenv0307's [proposal](https://github.com/Expensify/App/issues/16838#issuecomment-1503920274) to fix the style.

I do like @Prince-Mendiratta [proposal](https://github.com/Expensify/App/issues/16838#issuecomment-1503920275) but i don't agree it fixes a root problem. If this waa true - "To fix this and have common style across platforms" then I could agree with the proposal, but I'm pretty sure once you make those suggested changes the offline text would be vertically aligned only for ReportActionCompose but for any other screen in smallScreen devices the offline text won't be vertically aligned the same as ReportActionCompose. For example, i think it would be differently vertically aligned for report view and LHN view which would be quite weird IMO and then we would have to apply the same style changes for small screen devices. Let me know if i am mistaken.

**Comment by MelvinBot** [comment]
@mananjadhav @alexpensify @chiragsalian this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by alexpensify** [comment]
@Prince-Mendiratta - seeing if you have an update based on @chiragsalian's feedback, thanks for reviewing!

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by chiragsalian** [comment]
@Prince-Mendiratta, so I answered wrt your refactor suggestion over [here](https://github.com/Expensify/App/issues/16838#issuecomment-1509159670). Did you have the chance to read that? I felt like your refactor suggestion we still would not have common styles throughout the platform. It would remove the need for the extra check only in ReportActionCompose but i personally don't feel the refactor would be worth it for just that gain.

**Comment by chiragsalian** [comment]
Cool okay, Let's proceed with @dukenv0307 proposal. @dukenv0307 feel free to create the PR.

**Comment by MelvinBot** [comment]
📣 @dukenv0307 You have been assigned to this job by @chiragsalian!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by dukenv0307** [comment]
@mananjadhav @chiragsalian [The PR](https://github.com/Expensify/App/pull/17778) is ready for review

**Comment by alexpensify** [comment]
Thank you @dukenv0307 for the update!

**Comment by mananjadhav** [comment]
Just finished the checklist. @chiragsalian All yours.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.5-6 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17778


If no regressions arise, payment will be issued on 2023-05-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mananjadhav] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mananjadhav] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mananjadhav] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@mananjadhav] Determine if we should create a regression test for this bug.
- [ ] [@mananjadhav] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@alexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by alexpensify** [comment]
Do we have a regression test plan? If yes, share it here. Thanks!

**Comment by mananjadhav** [comment]
**Regression Test Proposal**

1. Open the App
2. Login with any account
3. Open any chat
4. Disable the internet connection
5. Verify that the message "You appear to be offline" is vertically center under compose box.

@chiragsalian @alexpensify are we fine with this :+1: or :-1: ?

**Comment by alexpensify** [comment]
Looks good, do you agree @chiragsalian?

**Comment by chiragsalian** [comment]
Sure, LGTM 👍

**Comment by alexpensify** [comment]
We have the following payouts. I see there is a 50% bonus because it was assigned on April 20 and was merged into production on April 24 (that's under the 3 business day mark):

Issue reporter: $250 - N/A (Internal) 
Contributor: $1,500 - @dukenv0307 
Contributor+: $1,500- @mananjadhav 

@chiragsalian and @mananjadhav  - Can you please 👍🏼 if you agree. Tomorrow, I'll complete the next steps in Upwork. Thank you!

**Comment by mananjadhav** [comment]
That's right @alexpensify

**Comment by alexpensify** [comment]
@mananjadhav - can you fill this out: https://github.com/Expensify/App/issues/16838#issuecomment-1522810111

I'm working on the payment process now and will follow up after.

**Comment by alexpensify** [comment]
Alright, the payment process has been started in Upwork. Please review and accept, I will continue with the next steps.

**Comment by mananjadhav** [comment]
Thanks @alexpensify.  I've already proposed the regression test proposal. I couldn't track the offending PR. @chiragsalian Were you able to ? `OfflineIndicator` has been through iterations and styling, when composer was added.

**Comment by alexpensify** [comment]
Alright, everyone has been paid out here via Upwork and I've closed the job. Great work!

**Comment by chiragsalian** [comment]
Nope, i couldn't track the offending PR. I didn't spend too much time on it either and i think its fine to proceed without it in this case.
